src/include/function/unary_operation_executor.h

```diff
@@ -53,8 +53,8 @@ struct UnaryOperationExecutor {
             auto pos = operand.state->selVector->selectedPositions[0];
             result.setNull(pos, operand.isNull(pos));
             if (!result.isNull(pos)) {
-                executeOnValue<OPERAND_TYPE, RESULT_TYPE, FUNC, OP_WRAPPER>(
-                    operand, pos, resultValues[pos], result);
+                executeOnValue<OPERAND_TYPE, RESULT_TYPE, FUNC, OP_WRAPPER>(operand, pos,
+                    resultValues[result.state->selVector->selectedPositions[0]], result);
```

Comment:
why are we changing this?

Reply:
This is a bug in the executor.
The result and source vector may not have the same state

Reply:
ok. Can u then in this function create local variable inputPos and resultPos and check unaryExecute/SelectSwitch as well as unary boolean executor?

I think we have the same problem across these two places